configure.ac

```diff
@@ -1671,6 +1671,37 @@ if test "x${je_cv_gcc_builtin_popcountl}" = "xyes" ; then
   AC_DEFINE([JEMALLOC_INTERNAL_POPCOUNTLL], [__builtin_popcountll], [ ])
 fi
 
+dnl Configure maximum number of tcaches/arenas
+AC_ARG_WITH([lg_tcache_limit],
+  [AS_HELP_STRING([--with-lg-tcache-limit=<lg-cache-limit>],
+  [Base 2 log of maximum number of tcaches (between 12 and 15). Trade off with number of arenas, which will be 2 ^ (24 - lg-cache-limit).])])
+
+if test "x$with_lg_tcache_limit" = "x"; then
+  # Default is 4096 tcaches, 4096 arenas
+  with_lg_tcache_limit=12
+fi
+
+if ! test "$with_lg_tcache_limit" -ge 12 -a "$with_lg_tcache_limit" -le 15; then
+  AC_MSG_ERROR([lg-cache-limit should be between 12 and 15.])
+fi
+
+dnl Internal macros
+tcache_and_arena_repr_bits=24
+tcache_repr_bits="$with_lg_tcache_limit"
+arena_repr_bits="$(($tcache_and_arena_repr_bits-$tcache_repr_bits))"
+tcache_shift=8
```

Comment:
I'd minimise the amount of maths we do in autoconf - it's not a particulary nice place to do that. 

Just expose the number of bits for tcache, and do all the other calculations in C.

Reply:
Agree about it not being nice, however, we need the arenas_all and arenas_destroyed to be literals. 

See the comment on:
https://github.com/couchbasedeps/jemalloc/blob/5.3.0-couchbase/include/jemalloc/jemalloc_macros.h.in#L44